proto/rill/runtime/v1/connectors.proto

```diff
@@ -197,3 +204,19 @@ message BigQueryListTablesResponse {
   string next_page_token = 1;
   repeated string names = 2;
 }
+
+message ScanConnectorsRequest {
+  string instance_id = 1;
+}
+
+message ScanConnectorsResponse {
+  repeated ScannedConnector connectors = 1;
+  map<string, string> existing_credentials = 2; // credentials already present in env file
+}
+
+message ScannedConnector {
+  string name = 1;
+  string type = 2;
+  repeated string variables = 3;
```

Comment:
1. Can't this be extracted from the connector spec already? Feels like we should either return the spec or have a separate API for getting the specs (don't we have that already?)
2. We should not conflate connector config with variables. An instance's *variables* are one way to set connector *config* (through the `connector.name.xxx` convention), but can also be used for other things (like templating into code files). And connector config can be set in other ways than variables, e.g. in `rill.yaml`.

Reply:
1. By spec do you mean only the keys and not values. We have one to list all keys, other to list all credentials for a connector. 
2. Sure we have some APIs a combination of which can give this entire information. I can remove these fields.